packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-customer-section/order-customer-section.tsx

```diff
@@ -0,0 +1,319 @@
+      />
+    </div>
+  )
+}
+
+const getCustomerName = (order: Order) => {
+  const { first_name: sFirstName, last_name: sLastName } =
+    order.shipping_address || {}
+  const { first_name: bFirstName, last_name: bLastName } =
+    order.billing_address || {}
+  const { first_name: cFirstName, last_name: cLastName } = order.customer || {}
+
+  const customerName = [cFirstName, cLastName].filter(Boolean).join(" ")
+  const shippingName = [sFirstName, sLastName].filter(Boolean).join(" ")
+  const billingName = [bFirstName, bLastName].filter(Boolean).join(" ")
+
+  const name = customerName || shippingName || billingName
+
+  return name
+}
+
+const ID = ({ order }: { order: Order }) => {
+  const { t } = useTranslation()
+
+  const id = order.customer_id
+  const name = getCustomerName(order)
+  const email = order.email
+  const fallback = (name || email).charAt(0).toUpperCase()
+
+  return (
+    <div className="text-ui-fg-subtle grid grid-cols-2 items-center px-6 py-4">
+      <Text size="small" leading="compact" weight="plus">
+        {t("fields.id")}
+      </Text>
+      <Link
+        to={`/customers/${id}`}
+        className="focus:shadow-borders-focus rounded-[4px] outline-none transition-shadow"
+      >
+        <div className="flex items-center gap-x-2 overflow-hidden">
+          <Avatar size="2xsmall" fallback={fallback} />
+          <Text
+            size="small"
+            leading="compact"
+            className="text-ui-fg-subtle hover:text-ui-fg-base transition-fg truncate"
+          >
+            {name || email}
+          </Text>
+        </div>
+      </Link>
+    </div>
+  )
+}
+
+const Company = ({ order }: { order: Order }) => {
+  const { t } = useTranslation()
+  const company =
+    order.shipping_address?.company || order.billing_address?.company
+
+  if (!company) {
+    return null
+  }
+
+  return (
+    <div className="text-ui-fg-subtle grid grid-cols-2 items-center px-6 py-4">
+      <Text size="small" leading="compact" weight="plus">
+        {t("fields.company")}
+      </Text>
+      <Text size="small" leading="compact" className="truncate">
+        {company}
+      </Text>
+    </div>
+  )
+}
+
+const Contact = ({ order }: { order: Order }) => {
+  const { t } = useTranslation()
+
+  const phone = order.shipping_address?.phone || order.billing_address?.phone
+  const email = order.email
+
+  return (
+    <div className="text-ui-fg-subtle grid grid-cols-2 items-start px-6 py-4">
+      <Text size="small" leading="compact" weight="plus">
+        {t("orders.customer.contactLabel")}
+      </Text>
+      <div className="flex flex-col gap-y-2">
+        <div className="grid grid-cols-[1fr_20px] items-start gap-x-2">
+          <Text
+            size="small"
+            leading="compact"
+            className="text-pretty break-all"
+          >
+            {email}
+          </Text>
+
+          <div className="flex justify-end">
+            <Copy content={email} className="text-ui-fg-muted" />
```

Comment:
todo: breaks out of the panel
![Screenshot 2024-03-01 at 12 47 07](https://github.com/medusajs/medusa/assets/16856471/e50f74bd-4ae6-4eed-8bc4-1811f9326a06)

Reply:
Could you check again, this should have been fixed already and looks good on my end

Reply:
Can you check again it should already be fixed, and it looks good on my end

Reply:
Looks good now 👍